plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/execute/SQLQueryJob.java

```diff
@@ -267,7 +266,8 @@ protected IStatus run(DBRProgressMonitor monitor)
                                 break;
                             case RETRY:
                                 // just make it again
-                                continue;
+                                this.schedule(100);
```

Comment:
Not clear process flow.
Retry must re-execute only the failed query, not entire script